packages/core/src/physics/DynamicCollider.ts

```diff
@@ -59,3 +273,35 @@ export class DynamicCollider extends Collider {
     this._updateFlag.flag = false;
   }
 }
+
+/**
+ * The collision detection mode constants.
+ */
+export enum CollisionDetectionMode {
+  /// Continuous collision detection is off for this dynamic collider.
+  Discrete,
+  /// Continuous collision detection is on for colliding with static mesh geometry.
+  Continuous,
+  /// Continuous collision detection is on for colliding with static and dynamic geometry.
+  ContinuousDynamic,
+  /// Speculative continuous collision detection is on for static and dynamic geometries
+  ContinuousSpeculative
+}
+
+/**
+ * Use these flags to constrain motion of dynamic collider.
+ */
+export enum DynamicColliderConstraints {
+  /** Freeze motion along the X-axis. */
```

Comment:
You can add `None=0`, We can constrain properties entirely using enum operations,not using number any more.

Reply:
+1